packages/bee-q/src/components/dropdown/bq-dropdown.tsx

```diff
@@ -0,0 +1,164 @@
+  // ========================
+
+  /** Distance (px) between the panel and the trigger element. */
+  @Prop({ reflect: true }) panelDistance?: number = 4;
+
+  /** Position of the panel */
+  @Prop({ reflect: true }) panelPlacement?: FloatingUIPlacement = 'bottom-start';
+
+  /**
+   * If true, panel is visible.
+   * You can toggle this attribute to show/hide the panel.
+   */
+  @Prop({ reflect: true }) panelOpen?: boolean = false;
+
+  /**
+   * Determines whether the scrollbar is visible or hidden within the panel.
+   */
+  @Prop({ reflect: true }) panelScrollbar?: boolean = false;
+
+  // Prop lifecycle events
+  // =======================
+
+  // Events section
+  // Requires JSDocs for public API documentation
+  // ==============================================
+
+  /**
+   * Handler to be called to check if the `bq-panel` switches state (visible/hidden).
+   * @returns CustomEvent - with value `{ opened: boolean }`
+   */
+  @Event() bqPanelChange: EventEmitter<{ opened: boolean }>;
+
+  // Component lifecycle events
+  // Ordered by their natural call order
+  // =====================================
+
+  componentDidLoad() {
+    this.initFloatingUIFromPanel();
+  }
+
+  // Listeners
+  // ==============
+
+  /** On click outside the panel */
+  @Listen('click', { target: 'document', passive: true })
+  onClickOutsidePanel(event: MouseEvent) {
+    // close the panel on click, except click within the panel or within the trigger element
+    if (!event.composedPath().includes(this.panelElement) && !event.composedPath().includes(this.trigger)) {
+      this.panelElement.open = false;
+    }
+  }
+
+  /**
+   * If the user press on Escape button
+   * or `<bq-dropdown>` loses focus
+   * then hide the panel
+   */
+  @Listen('keyup', { target: 'window', passive: true })
+  onEscape(event: KeyboardEvent) {
+    if (event.key === 'Escape' || (event.key === 'Tab' && !event.composedPath().includes(this.el))) {
+      this.panelElement.open = false;
+    }
+  }
+
+  @Listen('bqPanelVisibility', { passive: true })
+  onPanelVisibilityChange(event: CustomEvent) {
+    const isOpened: boolean = event.detail as boolean;
+
+    this.bqPanelChange.emit({ opened: isOpened });
+  }
+
+  @Listen('bqSelect', { passive: true })
+  onItemSelect() {
+    this.panelElement.open = false;
+  }
+
+  // Public methods API
+  // These methods are exposed on the host element.
+  // Always use two lines.
+  // Public Methods must be async.
+  // Requires JSDocs for public API documentation.
+  // ===============================================
+
+  // Local methods
+  // Internal business logic.
+  // These methods cannot be called from the host element.
+  // =======================================================
+
+  private openPanel = (): void => {
+    this.panelElement?.togglePanel();
+    this.panelElement.open = !this.panelElement.open;
+  };
+
+  private initFloatingUIFromPanel = (): void => {
+    this.panelElement?.setTriggerElement(this.el);
+  };
```

Comment:
Since the trigger element it will be passed through a property, we only need to handle the trigger element click to toggle the `open` property:

```suggestion
  private togglePanel = (): void => {
    this.open = !this.open;
  };
```